src/validator_main.cpp

```diff
@@ -222,13 +222,19 @@ int main(int argc, char** argv)
 
         if (path == ebi::vcf::STDIN) {
             BOOST_LOG_TRIVIAL(info) << "Reading from standard input...";
+            if (ebi::vcf::is_compressed_file(std::cin, path)) {
+                return 1;
+            }
             is_valid = ebi::vcf::is_valid_vcf_file(std::cin, path, validationLevel, ploidy, outputs);
         } else {
             BOOST_LOG_TRIVIAL(info) << "Reading from input file...";
             std::ifstream input{path};
             if (!input) {
                 throw std::runtime_error{"Couldn't open file " + path};
             } else {
+                if (ebi::vcf::is_compressed_file(input, path)) {
```

Comment:
note that `is_compressed_file` is consuming data from the stream, and in case the input is not compressed, the parser won't receive the first bits, thus giving false reports for any input file.

Reply:
look at valitador.hpp, there is a part of the code that copies the first line, maybe there is a better place to include this check. also it wouldn't be necessary to write the check twice, as in here.

Reply:
`is_compressed_file` is consuming data
I think we may use `seekg(0)` for that.

Reply:
Check is added twice due to the reason that once the input is in form of STDIN and once we are reading from file .
Even the `is_valid_vcf_file` check is used twice in this code.

Reply:
were you talking this about https://github.com/EBIvariation/vcf-validator/blob/abbc6fbe2fac9769d8c7c82daa756a66ea2ceba1/src/vcf/validator.cpp#L142
in https://github.com/EBIvariation/vcf-validator/pull/111#discussion_r166876466 "the code that copies first line."
I am thinking to use this `line` in `is_compressed_file` instead of `input` keeping this thing in mind that `seekg` may behave abnormally in future.